addon/components/o-s-s/nav-tab.ts

```diff
@@ -0,0 +1,48 @@
+import { assert } from '@ember/debug';
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+
+interface OSSNavTabArgs {
+  onSelection(): void;
+  tabArray: TabDefinition[];
+}
+
+interface TabDefinition {
+  icon?: string;
+  label?: string;
+  infoCircle?: boolean;
+  notificationDot?: boolean;
+  selected: boolean;
+  disabled: boolean;
+}
+
+export default class OSSNavTab extends Component<OSSNavTabArgs> {
+  constructor(owner: unknown, args: OSSNavTabArgs) {
+    super(owner, args);
+
+    assert(
+      '[component][OSS::NavTab] The parameter @onSelection of type function is mandatory',
+      typeof args.onSelection === 'function'
+    );
+
+    assert('[component][OSS::NavTab] The parameter @tabArray is mandatory', args.tabArray);
+
+    let tabArrayValide = true;
+
+    args.tabArray.forEach((element: TabDefinition) => {
+      if (!element.label && !element.icon) {
+        tabArrayValide = false;
+      }
+    });
```

Comment:
First, you can move this to a `_checkTabArrayValidity` private method
Second, this can be factorise into one line like:
```
!args.tabArray.some((element: TabDefinition) => !element.label && !element.icon)
```

The `Array#some` method returns true if one of the element respects the condition